hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/AvailableSpaceFilter.java

```diff
@@ -39,14 +39,11 @@ public class AvailableSpaceFilter implements Predicate<HddsVolume> {
 
   @Override
   public boolean test(HddsVolume vol) {
-    SpaceUsageSource usage = vol.getCurrentUsage();
-    long volumeCapacity = usage.getCapacity();
-    long free = usage.getAvailable();
+    StorageLocationReport report = vol.getReport();
+    long free = report.getRemaining();
     long committed = vol.getCommittedBytes();
     long available = free - committed;
-    long volumeFreeSpaceToSpare = vol.getFreeSpaceToSpare(volumeCapacity);
-    boolean hasEnoughSpace = VolumeUsage.hasVolumeEnoughSpace(free, committed,
-        requiredSpace, volumeFreeSpaceToSpare);
+    boolean hasEnoughSpace = VolumeUsage.hasVolumeEnoughSpace(report, requiredSpace);
 
     mostAvailableSpace = Math.max(available, mostAvailableSpace);
```

Comment:
Shall mostAvailableSpace consider to deduct volumeFreeSpaceToSpare too？

Reply:
Why not change the fullVolumes to Map<HddsVolume, Long> and save available directly?

Reply:
> Why not change the `fullVolumes` to `Map<HddsVolume, Long>` and save `available` directly?

I think providing more information about full volumes (free/used/committed/reserved/etc.) is useful to help understand why a volume does not have space.

Reply:
Okay, it makes sense.  Do you like to do the AvailableSpace removal in this patch?

Reply:
I would like to do it as part of HDDS-12721.